tommath.h

```diff
@@ -209,9 +209,9 @@ mp_err mp_init_size(mp_int *a, int size) MP_WUR;
 
 /* ---> Basic Manipulations <--- */
 #define mp_iszero(a) (((a)->used == 0) ? MP_YES : MP_NO)
-mp_bool mp_iseven(const mp_int *a) MP_WUR;
-mp_bool mp_isodd(const mp_int *a) MP_WUR;
 #define mp_isneg(a)  (((a)->sign != MP_ZPOS) ? MP_YES : MP_NO)
+#define mp_iseven(a) (((a)->used == 0) || (((a)->dp[0] & 1u) == 0u) ? MP_YES : MP_NO)
```

Comment:
that was a short period of time that they were functions... introduced for 1.2.0 and abandoned one day after 1.2.0 was out :rofl: see #142

Reply:
To be honest, I never was a fan of those and also not of the MP_IS_ZERO macros.
They exist only to make the fperrard linter happy but make the code worse. I mean this x?MP_YES:MP_NO pattern. I would use c99 stdbool.h instead.

Reply:
> I would use c99 stdbool.h instead

maybe it's time? We could start an issue on that topic!?
@nijtmans needs c89 as TCL needs to be compiled in VC6 for some ancient OS IIUC!?
@karel-m also had some reasons to have c89 code.
For how many more years have these legacy systems to be supported?

Just to make it clear: I could also live without this explicit compare dance, especially for `mp_bool`. How about switching to stdbool for c99 compilers and writing (and linting) the code against c99? But for now also still provide a fallback for c89. I'm aware that this is not 100% possible/compatible, but wouldn't it be an option!?

Reply:
Yes, this is what I am arguing all the time. It is no problem to provide stdbool.h for c89 too on vc6. But we could also just change the linting and avoid explicit compare for mp_bool. Then I would be happy, even if we don't use stdbool.

Reply:
> It is no problem to provide stdbool.h for c89

including this? ;-) 

```
mp_bool a = 1, b = 2;
if (a == b) printf("stdbool");
else printf("error");
```

> But we could also just change the linting rules and avoid explicit compare for mp_bool

@fperrad would that be acceptable?